Rocky.class.nut

```diff
@@ -199,51 +200,62 @@ class Rocky {
-
-                    // Pull out the parts as strings
-                    local headers = req.body.slice(hstart, bstart);
-                    local name = null;
-                    local filename = null;
-                    local type = null;
-                    foreach (header in split(headers, ";\n")) {
-                        local kv = split(header, ":=");
-                        if (kv.len() == 2) {
-                            switch (strip(kv[0]).tolower()) {
-                                case "name":
-                                    name = strip(kv[1]).slice(1, -1);
-                                    break;
-                                case "filename":
-                                    filename = strip(kv[1]).slice(1, -1);
-                                    break;
-                                case "content-type":
-                                    type = strip(kv[1]);
-                                    break;
-                            }
-                        }
-                    }
-                    local data = req.body.slice(bstart, fstart);
-                    local part = { "name": name, "filename": filename, "data": data, "content-type": type };
+            // Find the boundary in the contentType
+            local boundary;
+            local findString = regexp(@"boundary([ ]*)=");
+            local match = findString.search(contentType);
 
-                    parts.push(part);
-                }
-            } while (bindex != null);
+            if (match) {
+                boundary = contentType.slice(match.end);
+                boundary = strip(boundary);
+            } else {
+                throw "No boundary found in content-type";
+            }
+
+            // Remove all carriage returns from string (to support either \r\n or \n for linebreaks)
+            local body = "";
+            local bodyLines = split(req.body, "\r");
+            foreach (i, line in bodyLines) {
+                body += line;
+            }
+
+            // Find all boundaries in the body
+            local boundaries = [];
+            local bregex = regexp2(@"--" + boundary);
+            local bmatch = bregex.search(body);
+            while (bmatch != null) {
+                boundaries.push(bmatch);
+                bmatch = bregex.search(body, bmatch.begin+1);
+            }
+
+            // Create array of parts
+            for (local i = 0; i < boundaries.len() - 1; i++) {
+                local part = body.slice(boundaries[i].end + 1, boundaries[i+1].begin);
+
+                local partSplit = regexp2("\n\n").search(part);
+                local header = part.slice(0, partSplit.begin);
+                local data = part.slice(partSplit.end, -1);
+
+                // Get the name
+                local name = null;
+                local nameCapture = regexp2(@"(^|\W)name\s*\=\s*""([^""]*)""").capture(header);
```

Comment:
my feeling is that all these three (and a half) reg-exps can be reduced to just one, which would be much faster and smaller